spec/datadog/core/telemetry/emitter_spec.rb

```diff
@@ -74,6 +74,17 @@
         end
       end
     end
+
+    context 'with data' do
+      subject(:request) { emitter.request(request_type, data: data) }
+      let(:data) { { changes: ['test-data'] } }
+      let(:request_type) { 'app-client-configuration-change' }
+
+      it 'sends data to the emitter' do
+        request
+        expect(http_transport).to have_received(:request).with(payload: include('test-data'), request_type: anything)
+      end
```

Comment:
Should this test focus on testing that `data` gets propagated to `Datadog::Core::Telemetry::Event.new.telemetry_request`? If feels that checking against `http_transport` is skipping the real change on `Emitter`

Reply:
changed it.